clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -683,8 +685,8 @@ public void onSuccess(ClientResponse response, RequestFuture<Map<TopicPartition,
      * @param listOffsetResponse The response from the server.
      * @param future The future to be completed when the response returns. Note that any partition-level errors will
      *               generally fail the entire future result. The one exception is UNSUPPORTED_FOR_MESSAGE_FORMAT,
-     *               which indicates that the broker does not support the v1 message format and results in a null
-     *               being inserted into the resulting map.
+     *               which indicates that the broker does not support the v1 message format and causes a null value
```

Comment:
Hmm.. that's a fair point. Maybe it won't work. It's a little confusing because we don't have a null check for the timestamp. I guess we assume that the timestamp will not be null because we have are ensured to be dealing with a v1 (or later) ListOffset request. It might be worth adding a comment somewhere so that expectation is not hidden.